clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1921,21 +1909,40 @@ protected boolean checkAndLogIfLagIsAcceptableForHybridStore(
     boolean isLagAcceptable = offsetLag <= offsetThreshold;
 
     if (shouldLogLag) {
-      LOGGER.info(
-          "{} [{} lag] partition {} is {}lagging. {}Lag: [{}] {} Threshold [{}]",
-          isOffsetBasedLag ? "Offset" : "Time",
-          consumerTaskId,
+      logLag(
+          isOffsetBasedLag,
           pcs.getPartition(),
-          (isLagAcceptable ? "not " : ""),
-          (isOffsetBasedLag ? "" : "The latest producer timestamp is " + latestConsumedProducerTimestamp + ". "),
+          isLagAcceptable,
+          latestConsumedProducerTimestamp,
           offsetLag,
-          (isLagAcceptable ? "<" : ">"),
-          offsetThreshold);
+          offsetThreshold,
+          "");
     }
 
     return isLagAcceptable;
   }
 
+  protected void logLag(
+      boolean isOffsetBasedLag,
+      int partition,
+      boolean isLagAcceptable,
+      long latestConsumedProducerTimestamp,
+      long offsetLag,
+      long offsetThreshold,
+      String lagLogFooter) {
+    LOGGER.info(
+        "{} [{} lag] partition {} is {}lagging. {}Lag: [{}] {} Threshold [{}]{}",
```

Comment:
nit: can we put spaces between `{}` and messages? also `logLag` sounds bit funny, maybe rename to something like `logOffsetLag` ?

Reply:
In the final string there is nothing stuck to anything else. That being said, I refactored the formatting a bit to minimize cases where `{}` is stuck to the next token...

The function is also used for time lag, so putting offset in the name seems wrong to me... but I'm open to other name suggestions if any.